src/libsystemd/sd-netlink/netlink-types-rtnl.c

```diff
@@ -839,8 +841,8 @@ static const NLType rtnl_proto_down_reason_types[] = {
 DEFINE_TYPE_SYSTEM(rtnl_proto_down_reason);
 
 static const NLType rtnl_link_types[] = {
-        [IFLA_ADDRESS]             = { .type = NETLINK_TYPE_ETHER_ADDR },
-        [IFLA_BROADCAST]           = { .type = NETLINK_TYPE_ETHER_ADDR },
+        [IFLA_ADDRESS]             = { .type = NETLINK_TYPE_ETHER_ADDR, .size = ETH_ALEN },
+        [IFLA_BROADCAST]           = { .type = NETLINK_TYPE_ETHER_ADDR, .size = ETH_ALEN },
```

Comment:
These depends on the hardware, e.g. INFINIBAND_ALEN for infiniband interfaces. Please drop the change.

Reply:
dropped